src/python_testing/TC_OCC_2_2.py

```diff
@@ -0,0 +1,113 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OCC_2_2(MatterBaseTest):
+    async def read_occ_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.OccupancySensing
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_OCC_2_2(self) -> str:
+        return "[TC-OCC-2.2] OccupancySensorTypeBitmap and OccupancySensorType interdependency with server as DUT"
+
+    def steps_TC_OCC_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read OccupancySensorType attribute with repect to featureMap."),
+            TestStep(3, "Read OccupancySensorTypeBitmap attribute with repect to featureMap.")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_2(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        feature_map_PIR = Clusters.OccupancySensing.Bitmaps.Feature.kPassiveInfrared
+        feature_map_US = Clusters.OccupancySensing.Bitmaps.Feature.kUltrasonic
+        feature_map_PHY = Clusters.OccupancySensing.Bitmaps.Feature.kPhysicalContact
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+        
+        self.step(2)      
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+            feature_map = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.attributes.FeatureMap)
+            
+            if (feature_map != feature_map_PIR)&(feature_map != feature_map_US)&(feature_map != feature_map_PHY):

```

Comment:
feature map is a bitmap. You are not supposed to use equality on bitmaps. You can have something like:

```py
is_PIR = (feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kUltrasonic) != 0
is_PHY = (feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kPhysicalContact) != 0
```

and use those booleans.

Reply:
Anything else seems to complex. In practice: I cannot review the if/elif blocks below for correctness (and unfortunately test plan seems to be very dismissinve in `Verify the value of OccupancySensorTypeBitmap and OccupancySensorType are aligned` .... if we can we should expand on that.

Reply:
also step 2 says `Verify OccupancySensorType is set to one of the non-reserved values: 0-PIR, 1-Ultrasonic, 2-PIR and ultrasonic, and 3-Physical contact.` ... this feature_map is unclear here. We should just verify the sensor type for set to one of the known values (we do not accept unknown). Sounds like we want a `assert_in` and no feature map dependency.

The complexity comment does apply to the one below.

Reply:
Will triage and resolve by the issue filed. (https://github.com/project-chip/connectedhomeip/issues/34686)